### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by shyusu4**
## PR Description

- Implemented composition (as an example of the design pattern).
- Ran a program using the command line.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  def correct_name
+    @name
+  end
+
+  def can_use_services?
+    @parent_permission || is_of_age?
+    false
+  end
```
Comment (HAMMAS-SALEEM) [inline person.rb:30]: - [ ] You have successfully implemented the methods but i guess you have implemented them in the private section 😄 means the **correct_name** and **can_use_services?** methods should be out of the private section means either add **public** keyword at line 26 or put these two functions before line 15 👍

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @shyusu4 :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following GitFlow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (shyusu4) [reply person.rb:30]: Hi👋 thanks for the review, I'll do that.

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @shyusu4,

✔️ Implementation of class decoration

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**